gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/main/java/io/gravitee/apim/core/documentation/use_case/ApiCreateDocumentationPageUseCase.java

```diff
@@ -77,18 +76,27 @@ public record Output(Page createdPage) {}
     private void validateParentId(Page page) {
         var parentId = page.getParentId();
 
-        if (Objects.nonNull(parentId) && !parentId.isEmpty()) {
-            var foundParent = pageCrudService.findById(parentId);
-
-            if (foundParent.isPresent()) {
-                if (!foundParent.get().isFolder()) {
-                    throw new InvalidPageParentException(parentId);
-                }
-                return;
-            }
+        if (parentId == null || parentId.isEmpty()) {
+            page.setParentId(null);
+            return;
         }
 
-        page.setParentId(null);
+        pageCrudService
+            .findById(parentId)
+            .ifPresentOrElse(
+                p -> {
+                    if (!p.isFolder()) {
+                        throw new InvalidPageParentException(page.getParentId());
+                    }
+                    this.apiDocumentationDomainService.validateNameIsUnique(
```

Comment:
I think that having this check in another method after the validateParentId would be a bit clearer. Placing it in the `validateParentId` method wraps it up into the parentId check, when name uniqueness is a different business rule.